### PR TITLE
First update dockeragent [draft]

### PR DESCRIPTION
**Pull request by AnjaBruls**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+aiohttp
```
Comment (joente) [inline requirements.txt:1]: Maybe mark aiohttp to a specific version?

File: dockeragent.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    args = parser.parse_args()
+
+    cl = Agent(
+        'dockeragent',
+        __version__,
+        CHECKS,
+    )
+
+    cl.setup_logger(args.log_level, args.log_colorized)
```
Comment (joente) [inline dockeragent.py:35]: Move `setup_logger` before the agent init as this allows logging in the init. I think I would not even chose to make the setup_logger part of the agent class.

File: lib/agent.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            if getattr(c, 'required', False)
+        ]
+
+        self.token = os.getenv('OSDA_TOKEN', None)
+        self.environment_uuid = int(os.getenv('OSDA_ENVIRONMENT_UUID', '0'))
+        self.host_uuid = int(os.getenv('OSDA_HOST_UUID', '0'))
+        self.api_uri = os.getenv('OSDA_API_URI', None)
+
+    @staticmethod
+    def setup_logger(log_level: str = 'warning', log_colorized: bool = False):
```
Comment (joente) [inline lib/agent.py:34]: Maybe just remove this method?

File: requirements.txt (hunk 1 above)
Comment (AnjaBruls) [reply requirements.txt]: Update in [13e4c34](https://github.com/oversight/dockeragent/pull/2/commits/13e4c348464d19411e8e71efba2f3e0d9b81d50d)

File: dockeragent.py (hunk 2 above)
Comment (AnjaBruls) [reply dockeragent.py]: See [13e4c34](https://github.com/oversight/dockeragent/pull/2/commits/13e4c348464d19411e8e71efba2f3e0d9b81d50d)

File: lib/agent.py (hunk 3 above)
Comment (AnjaBruls) [reply lib/agent.py]: See [13e4c34](https://github.com/oversight/dockeragent/pull/2/commits/13e4c348464d19411e8e71efba2f3e0d9b81d50d)

File: dockeragent.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+if __name__ == '__main__':
+    setproctitle('dockeragent')
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '-l', '--log-level',
+        default='warning',
+        help='set the log level',
```
Comment (joente) [inline dockeragent.py:19]: set the log lever or use the environment variable OS_LOG_LEVEL

File: dockeragent.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+if __name__ == '__main__':
+    setproctitle('dockeragent')
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '-l', '--log-level',
+        default='warning',
+        help='set the log level or use the environment variable OS_LOG_LEVEL',
```
Comment (joente) [inline dockeragent.py:19]: Change to OSDA_LOG_LEVEL

**Review by joente — APPROVED** [review]

**Pull request merged**